src/Couchbase.Lite.Shared/API/Database/Database.cs

```diff
@@ -228,7 +225,33 @@ public string Path
         [@NotNull]
         internal ThreadSafety ThreadSafety { get; } = new ThreadSafety();
 
-        private bool IsShell { get; }
+        internal bool IsClosedLocked
```

Comment:
This is never used except in tests?  What is it for?

Reply:
yes, its purpose of existence is for testing. iOS has it in their test...

Reply:
Why is it needed just for a test?

Reply:
FYI I talked to @pasin and he will remove it from iOS

Reply:
https://issues.couchbase.com/browse/CBL-892

Reply:
strange, I do see iOS is using `Database.isClosedLocked()` in their unit tests.
for ex: https://github.com/couchbase/couchbase-lite-ios/commit/f83efb4b6d01c525982b9f38acb953b2d34bcd03#diff-124d009e59a6f172bad43ef7c3856c78R1795

Reply:
It's optional. So please decide whether to have it or not based on your implementation and test. I will let @jayahariv double check this on iOS.